### PR TITLE
Wrap C++ strings returned to R in Rcpp::String to prevent UTF-8 mangling on Windows

### PR DESCRIPTION
**Pull request by knapply**

### CODE REVIEW

**Comment by ChrisMuir** [comment]
One thing to call out, there are a few other spots that use `.GetString()`, that I believe will also need to be wrapped in `Rcpp::String()` (Brendan correct me if I'm wrong):

https://github.com/knapply/jsonify/blob/master/inst/include/jsonify/from_json/simplify/simplify.hpp#L109

https://github.com/knapply/jsonify/blob/master/src/pretty.cpp#L22

https://github.com/knapply/jsonify/blob/master/src/pretty.cpp#L37

https://github.com/knapply/jsonify/blob/master/src/pretty.cpp#L50

**Comment by knapply** [comment]
@ChrisMuir Good calls. I wrapped all but https://github.com/knapply/jsonify/blob/master/src/pretty.cpp#L50 as I cannot figure out how to get `Rcpp::Rcout` to not use native encoding. That said `jsonify::pretty_json()` works completely fine.

I was encountering the same mangling making round trips, so I also modified `finalise_json()` here: https://github.com/SymbolixAU/jsonify/blob/473969ded465f99ac7d78cd3ad1189f6a431b627/inst/include/jsonify/to_json/utils.hpp#L34

**Comment by ChrisMuir** [comment]
👍 Looks good to me!

File: tests/testthat/test-from_json.R [hunk 1]
```diff
@@ -421,15 +421,33 @@ test_that("round trips work", {
 })
 
 test_that("UTF-8 encoding is not mangled", {
-  skip_if_not(Sys.info()['sysname'] == "Windows")
+  skip_if_not(Sys.info()["sysname"] == "Windows")
```
Comment (dcooley) [inline tests/testthat/test-from_json.R]: I think we can remove this check because we want these tests to succeed on all platforms.

**Comment by knapply** [comment]
Apologies, I screwed up by selecting master for a merge, and get the following warning when I look at changing it to the Issue 56 branch: "Some commits from the old base branch may be removed from the timeline, and old review comments may become outdated."

Obviously I'll defer to you all on what you'd like me to do there.

Slightly related to @dcooley's review, I can expand the `/.travis.yml` coverage (specifically more R versions and iOS coverage) if desired. Unfortunately Travis doesn't yet support R-on-Windows testing.

**Comment by SymbolixAU** [comment]
@knapply no worries - It all seems to have merged correctly.

**Comment by SymbolixAU** [comment]
And I'll defer to you regarding the `/.travis.yml` - I don't really know what a good setup is for testing :)

**Pull request merged**